### PR TITLE
feat(seed): initial implementation of mojaloop switch seeder

### PR DESCRIPTION
**Pull request by lewisdaly**
- Implements a new, faster, easier to configure and more readable method for seeding 
- The config is currently hardcoded for the `docker-live` environment we have specified here, but there is no reason it can't be expanded upon and used in the `docker-local` or indeed other mojaloop environments

### CODE REVIEW

File: src/seed/runResult.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+}
+
+export interface FailureResult {
+  type: RunResultType.FAILURE
+  warnings: Array<string>
+  errors: Array<Error>
+}
+
+export class Result {
+  public static makeSuccessResult(warnings?: Array<string>): SuccessResult {
```
Comment (eoln) [inline src/seed/runResult.ts:21]:
```suggestion
  public static Success(warnings?: Array<string>): SuccessResult {
```

maybe let short the names of these static methods?

File: src/seed/runResult.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    return {
+      type: RunResultType.SUCCESS,
+      warnings: warnings || []
+    }
+  }
+
+  public static makeFailureResult(errors: Array<Error>, warnings?: Array<string>): FailureResult {
```
Comment (eoln) [inline src/seed/runResult.ts:28]:
```suggestion
  public static Failure(errors: Array<Error>, warnings?: Array<string>): FailureResult {
```

File: src/seed/requests.ts [hunk 3]
```diff
@@ -0,0 +1,261 @@
...
+    return this.executeRequest(options)
+  }
+
+  public static async postALSParticipant(host: string, request: PostALSParticipantRequest): Promise<AxiosResponse<any>> {
+    const url = `${host}/participants/${request.idType}/${request.idValueOrSubType}` + (request.idValue ? `/${request.idValue}` : '')
+    const options: AxiosRequestConfig = {
+      method: 'post',
+      url,
+      headers: {
+        'Accept': 'application/vnd.interoperability.participants+json;version=1',
```
Comment (eoln) [inline src/seed/requests.ts:228]: maybe we should take these headers (Accept, Content-Type) values from config, not hardcode here?

**Review by eoln — DISMISSED** [review]
lgtm, some comments

File: src/seed/index.ts [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+import makePartySteps from './steps/partySteps'
+import { SeedCollection } from './types'
+
+
+const collections: Array<SeedCollection> = [
+  hubSteps(config),
+  oracleSteps(config),
+
+  // Generate a set of participant steps for each participant
+  ...config.participants.map(p => makeParticipantSteps(p as unknown as PISPParticipant | DFSPParticipant)(config)),
```
[the remark this answers is not in the document]
Comment (lewisdaly) [reply src/seed/index.ts:15]: TS was getting confused by the union types - let me try again since I think this was an older implementation.

**Review by eoln — APPROVED** [review]
lgtm
